### ISSUE TITLE
Some UI apps would like to exit when last window closes (on Mac)

### ISSUE THREAD

**Issue by mrjbq7** [factor/factor#1746]
I think we should support this.  Some apps (like ``spheres`` or ``bunny`` or ``boids``) are essentially single-window apps and would probably like to exit when the last window is closed.

We can do this by overriding this to return ``YES`` instead of ``NO`` (the default):

```
-(BOOL)applicationShouldTerminateAfterLastWindowClosed:(NSapplication*)sender { return YES; }
```

But, we don't want to do this for all UI apps, so need a way to indicate such...

**Comment by bjourne** [comment]
Is that a mac-specific problem? They terminate fine when you run them like this `./factor -run=spheres`

**Comment by mrjbq7** [comment]
Seems like it's just a Mac thing.  GTK and Windows both close after the last window is closed by default.

I could imagine wanting the default Factor app (``ui.tools``) to still keep running when you close the last window (since you can make a new listener from the menu bar and that's more "Mac-like"), but most other apps may not want that.

**Comment by mrjbq7** [comment]
Kinda weird that almost all of our ``deploy.factor`` files have ``{ "stop-after-last-window?" t }`` in them... but that doesn't seem to be used anywhere.

**Comment by bjourne** [comment]
I think the difference is that the gtk and windows backends naturally jumps out of the loops when there are no more windows. If you look at the gtk backend it has `[ [ gtk_main ] with-timer ] with-event-loop` which means it can be jumped out of. The cocoa backend has nothing similar.

**Comment by mrjbq7** [comment]
The way cocoa UI does it (I believe) is through that overridden method I mentioned above. It might be worth returning the value of a symbol and have the `ui.tools` set to NO but default to YES for other apps. And cleanup those deploy files.

**Comment by mrjbq7** [comment]
Fixed (at least for macOS) in 96a0c30baf807ecbc268fe23035d16014350e250 and 8b455d2306df97e41f7b52d7a0c355b41a709abe.
